### DISCUSSION TITLE
Implementing Built-in Rate Limiting functionality for Flask API development

### DISCUSSION THREAD

**Discussion by PythonHacker24** [pallets/flask#5228] [Ideas]
Hi, this is Aditya and I have come up with the idea of implementing a rate-limiting feature for the flask. Third-party rate-limiting modules exist but that's not the most secure way to get the job done. I would like to start working with an inbuilt functionality, that would allow developers to implement a rate-limiting to the API server allowing smooth functionality of the server and protection from DDOS attacks on the server. I would like to have recommendations on this topic, whether it is feasible, want to join me in making this idea come real, etc. 

P.S: Considering the primary feature of Flask to be a lightweight and fast framework, this implementation has to be minimal. It would be great if I get to know what more constraints are to be considered before developing the functionality.

**Comment by RonnyPfannschmidt** [comment]
As flask should always run behind some type of reverse proxy,load balancer/static file serving and SSL terminating Ingress that typically already handles load and rate limits, it seems misplaced in flasks

Additionally topic's like ddos mitigation are different beast's than rate limits

So the different use cases ought to be clearly positioned on their own before mixing them up

**Reply by PythonHacker24** [reply]
I think your explanation is acceptable and I understand the very first flaw in the idea.
